libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/MockLocationUpdatesRule.kt

```diff
@@ -0,0 +1,111 @@
+package com.mapbox.navigation.testing.ui
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.SystemClock
+import android.util.Log
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import java.util.Date
+
+private const val mockProviderName = LocationManager.GPS_PROVIDER
+
+/**
+ * Rule that sets up a mock location provider that can inject location samples
+ * straight to the device that the test is running on.
+ */
+class MockLocationUpdatesRule : TestWatcher() {
+
+    private val instrumentation = getInstrumentation()
+    private val appContext = (ApplicationProvider.getApplicationContext() as Context)
+
+    private val locationManager: LocationManager by lazy {
+        (appContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager)
+    }
+
+    override fun starting(description: Description?) {
+        with(instrumentation.uiAutomation) {
+            val result = executeShellCommand(
+                "appops set " +
+                    appContext.packageName +
+                    " android:mock_location allow"
+            )
+            result.close()
+            // gives the adb command chance to process
+            // and avoids occasional issues with registering the mock provider
+            Thread.sleep(1000)
```

Comment:
Wondering if we could workaround / alleviate this 1s wait somehow 🤔

Reply:
I tried to read the returned stream but it's always empty. Couldn't find another way.